src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLContext.m

```diff
@@ -0,0 +1,470 @@
+                                                      linear:linear
+                                                 cycleMethod:cycleMethod
+                                                    numStops:numStops
+                                                         m00:m00
+                                                         m01:m01
+                                                         m02:m02
+                                                         m10:m10
+                                                         m11:m11
+                                                         m12:m12
+                                                      focusX:focusX
+                                                   fractions:fractions
+                                                      pixels:pixels] autorelease];
+}
+
+- (void)setTexturePaint:(jboolean)useMask
+                pSrcOps:(jlong)pSrcOps
+                 filter:(jboolean)filter
+                    xp0:(jdouble)xp0
+                    xp1:(jdouble)xp1
+                    xp3:(jdouble)xp3
+                    yp0:(jdouble)yp0
+                    yp1:(jdouble)yp1
+                    yp3:(jdouble)yp3
+{
+    BMTLSDOps *srcOps = (BMTLSDOps *)jlong_to_ptr(pSrcOps);
+
+    if (srcOps == NULL || srcOps->pTexture == NULL) {
+        J2dRlsTraceLn(J2D_TRACE_ERROR, "MTLContext_setTexturePaint: texture paint - texture is null");
+        return;
+    }
+
+    J2dTraceLn1(J2D_TRACE_INFO, "MTLContext.setTexturePaint [tex=%p]", srcOps->pTexture);
+
+    self.paint = [[[MTLTexturePaint alloc] initWithUseMask:useMask
+                                                textureID:srcOps->pTexture
+                                                  isOpaque:srcOps->isOpaque
+                                                   filter:filter
+                                                      xp0:xp0
+                                                      xp1:xp1
+                                                      xp3:xp3
+                                                      yp0:yp0
+                                                      yp1:yp1
+                                                      yp3:yp3] autorelease];
+}
+
+- (id<MTLCommandBuffer>)createCommandBuffer {
+    return [self.commandQueue commandBuffer];
+}
+
+/*
+ * This should be exclusively used only for final blit
+ * and present of CAMetalDrawable in MTLLayer
+ */
+- (id<MTLCommandBuffer>)createBlitCommandBuffer {
+    return [self.blitCommandQueue commandBuffer];
+}
+
+-(void)setBufImgOp:(NSObject*)bufImgOp {
+    if (_bufImgOp != nil) {
+        [_bufImgOp release]; // context owns bufImgOp object
+    }
+    _bufImgOp = bufImgOp;
+}
+
+-(NSObject*)getBufImgOp {
+    return _bufImgOp;
+}
+
+@end
+
+/*
+ * Class:     sun_java2d_metal_MTLContext
+ * Method:    getMTLIdString
+ * Signature: ()Ljava/lang/String;
+ */
+JNIEXPORT jstring JNICALL Java_sun_java2d_metal_MTLContext_getMTLIdString
```

Comment:
dead code?

Reply:
Looks like removing it under JDK-8260640